CppCoreGuidelines.md

```diff
@@ -20029,9 +20029,95 @@ Following this rule leads to weaker invariants,
 more complicated code (having to deal with semi-constructed objects),
 and errors (when we didn't deal correctly with semi-constructed objects consistently).
 
-##### Example
+##### Example, bad
 
-    ???
+    class Picture
+    {
+        int mx;
+        int my;
+        char * data;
+    public:
+        Picture(int x, int y)
+        {
+            mx = x,
+            my = y;
+            data = nullptr;
+        }
+    
+        ~Picture()
+        {
+            Cleanup();
+        }
+    
+        bool Init()
+        {
+            // invariant checks
+            if (mx <= 0 || my <= 0 || mx % 2 != 0 || my % 2 != 0) {
+                return false;
+            }
+            if (data) {
+                return false;
+            }
+            data = (char*) malloc(x*y*sizeof(int));
+            return data != nullptr;
+        }
+    
+        void Cleanup()
+        {
+            if (data) free(data);
+            data = nullptr;
+        }
+    };
+    
+    Picture picture(3, 5); // not ready-to-use picture here
+    // this will fail..
+    if (!picture.Init()) {
+        puts("Error, invlaid picture");
+    }
+    // now have a invalid picture object instance.
+
+##### Example, good
+
+    class Picture
+    {
+        int mx;
+        int my;
+        vector<char> data;
+    
+        inline int check_size(int s)
+        {
+            // invariant check
+            if (s <= 0  || s % 2 != 0) {
+                throw std::invalid_argument("wrong size for picture");
+            }
+            return s;
+        }
+    
+    public:
+        Picture(int x, int y)
+            : mx(check_size(x))
```

Comment:
Expects (I.6).. or even a better type than "int" (I.4) could make it nicer, though I think it would still be contrived (what are those units that must be even for a valid Picture?)

Reply:
As far as I know, Expects does not return something, so I cant use it for member initialization.
I agree, the 2 ints as parameter are not the best choice. The focus of the bad and good example is on the two phase initialization with the Init() method and the problems arising from it.
I made the check for "even" just as an example. Some (probably older) picture formats have this requirements.
Maybe it is a good change, when I use the Expects() macro inside the check_size() helper function.
Furthermore to make the example more generic, I will remove the check for even.
A simple Size2D class with width and height instead of the ints could be another improvement.

Reply:
I made some aprovements. Please let me know, what do you think about it now.
As I thought about to change the 2 parameter for x and y to one e.g. Size2D class, I realized that this will not solve the issue completely because the constructor of Size2D will still need 2 parameter for width and height, so I just changed the type from int to size_t for now

Reply:
Sorry, I am not familiar yet with the common procedure on github. Do I need to resolve conversation? 
I think, I changed it enough for a new review. So, I will use the resolve conversation.... Otherwise a new comment can be opened.